src/client/mod.rs

```diff
@@ -429,8 +460,8 @@ pub enum ClientError {
     Digest(SignerError),
     #[error("failed to serialize signature as header value, {0}")]
     SerializeHeaderValue(header::InvalidHeaderValue),
-    #[error("failed to parse url endpoint, {0}")]
-    ParseUrlEndpoint(url::ParseError),
+    #[error("too many headers")]
```

Comment:
Could you add a more explicit error message.

Reply:
The actual limit is not publicly exposed by reqwest and might changed without notice. Currently, it's `32768` entries.